### ISSUE TITLE
FITS column name becomes case-sensitive when read in as QTable

### ISSUE THREAD

**Issue by pllim** [astropy/astropy#16221]
I understand how this happened but it seems a little counter-intuitive because QTable does not allow case-insensitivity while FITS does, causing conflicting behavior via Unified I/O. What is the best way to work around this?

```python
import numpy as np
from astropy import units as u
from astropy.io import fits
from astropy.table import QTable

pix = np.arange(5, dtype=int) + 1
wav = (pix * 0.1) * u.micron
trace = np.array([0, 0.5, 1, 0.9, 0])
coldefs = fits.ColDefs([
    fits.Column(name="X", format="I", array=pix),
    fits.Column(name="WAVELENGTH", format="E",
                unit=wav.unit.to_string(format="fits"), array=wav.value),
    fits.Column(name="TRACE", format="E", array=trace)])
hdulist = fits.HDUList([
    fits.PrimaryHDU(),
    fits.BinTableHDU.from_columns(coldefs)])

t = QTable.read(hdulist)
```

```python
>>> t["trace"]
KeyError: 'trace'
>>> hdulist[1].data["trace"]
array([0. , 0.5, 1. , 0.9, 0. ], dtype=float32)
```

**Comment by mhvk** [comment]
Hmm, just for reference in discussion, fits uses: https://github.com/astropy/astropy/blob/96cc7fbefd59e79d096802efe9f75b2f1d042487/astropy/io/fits/column.py#L2116-L2137

So, one option would be for [Q]Table to have something similar. I'm not sure we want to follow that, though... But another option might be to have a way for `QTable` to change all names to lower (or upper) case (perhaps as an argument in `.read()`?). Would that help? In principle, one could even keep the original name in the metadata, if one worried about round-tripping. Or perhaps one could set aliases for column names?

Let's ping @taldcroft too!

**Comment by pllim** [comment]
I ended up post-processing my data like this:

```python
colnames_lower = [c.lower() for c in t.colnames]
input_col_lower = input_col.lower()
the_col_i_want = t.columns[colnames_lower.index(input_col_lower)]
```

Also, this really is just for FITS. I would expect a normal Python-like table to be case-sensitive.

**Comment by pllim** [comment]
Not sure what the ideal solution here would be. I am okay if we close this as won't fix too. Just caught be off-guard.

**Comment by taldcroft** [comment]
I thought a little bit about what it would take for case-insensitive column names in a Table subclass. The first-order stuff like column access would be straightforward, but I think it would be a pretty big mess to get everything (e.g. table operations) all working consistently. So probably the best solution is post-processing and I would suggest closing this issue.

Another workaround is a single post-process step after reading that makes all the column names uniform:
```
t.rename_columns(t.colnames, [c.lower() for c in t.colnames])
```
After that always use lower-case names.

**Comment by mhvk** [comment]
Indeed, post-processing the names would seem the safest. I guess the remaining question then is whether it would be helpful to have an argument in `.read()` for doing that. And of course how best to document this...

**Comment by taldcroft** [comment]
This could be in the FITS Table reader and should show up in `Table.read.help(format='fits')`. Now what fraction of users know about that is a different story... And of course the narrative Unified I/O docs for FITS is another place.

**Comment by pllim** [comment]
> After that always use lower-case names.

That does not achieve parity with FITS interface though. The problem I actually encountered was that I had custom code to read FITS table using only `io.fits` API but I couldn't take advantage of unit support from astropy so I switched to QTable, but after the switch, we found out that people have not been making their FITS columns consistently cased throughout the database, which isn't a problem if you assume case-insensitivity. Anyways, I will keep this open in case you want to add doc, otherwise feel free to close as won't fix. Thanks for the feedback!
